### PR TITLE
Add architecture to node repo rest api

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/FlavorConfigBuilder.java [hunk 1]
```diff
@@ -20,14 +20,33 @@ public FlavorsConfig build() {
...
     public FlavorsConfig.Flavor.Builder addFlavor(String flavorName,
-                                                  double cpu, double mem, double disk, double bandwidth,
-                                                  boolean fastDisk, boolean remoteStorage,
-                                                  Flavor.Type type) {
+                                                  double cpu,
+                                                  double mem,
+                                                  double disk,
+                                                  double bandwidth,
+                                                  Flavor.Type type,
+                                                  String architecture) {
```
Comment (bjorncs) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/FlavorConfigBuilder.java:37]: Consider using enum as parameter type

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/FlavorConfigBuilder.java [hunk 2]
```diff
@@ -20,14 +20,33 @@ public FlavorsConfig build() {
...
+
+    public FlavorsConfig.Flavor.Builder addFlavor(String flavorName,
+                                                  double cpu,
+                                                  double mem,
+                                                  double disk,
+                                                  double bandwidth,
+                                                  boolean fastDisk,
+                                                  boolean remoteStorage,
+                                                  Flavor.Type type,
+                                                  String architecture) {
```
Comment (bjorncs) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/FlavorConfigBuilder.java:49]: Consider using enum as parameter type

**Review by bjorncs — APPROVED** [review]

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/FlavorConfigBuilder.java (hunk 1 above)
Comment (hmusum) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/FlavorConfigBuilder.java]: Fixed

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/FlavorConfigBuilder.java (hunk 2 above)
Comment (hmusum) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/FlavorConfigBuilder.java]: Fixed

**Pull request merged**